.github/workflows/post-pr-comment.yml

```
@@ -0,0 +1,29 @@
+name: Post pull-request comment
+on:
+  workflow_run:
+    workflows: ["Query help preview"]
+    types:
+      - completed
+
+permissions:
+  pull-requests: write
+
+jobs:
+  post_comment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifact
+        run: gh run download "${{ github.event.workflow_run.id }}" --repo "${{ github.repository }}" --name "comment"
```

Nit. The following avoids dynamically generating bash code, and it works in the degenerate case where quotes can appear `.id` or .repository`. 
```suggestion
        run: gh run download "${RUN}" --repo "${REPO}" --name "comment"
        env: 
          RUN: ${{ github.event.workflow_run.id }}
          REPO: ${{ github.repository }}
```

(similar changes can be applied below)

---

Done!